### ISSUE TITLE
Refrigerator support and how is Nature Remo-E (lite)?

### ISSUE THREAD

**Issue by ahori** [scottyphillips/echonetlite_homeassistant#182]
I have a Nature Remo-E lite, a Mitsubishi fridge, and a Mitsubishi Air conditioner.  I recently found your echonetlite HACS and installed into my home assitant. Those three device are found, but it seems that your implementation does not support Nature Remo-E lite and the fridge, but the  are conditioner. 

Is it possible to implement these two devices if I could give you the information needed and could be the tester of those devices? 

I am sorry, I am not familiar with the home assistant implementation and it would be quite hard for me to implement the devices without any help. Things could go worse because I have been (mostly) C programmer and I have only little experiences with Python, JSON, and others, but C. 

Here is the air conditioner device info;
![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/d57fb0a0-f4c1-4968-b05f-80ef071d5334)

the fridge; 
![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/f39e8684-20b8-40f0-886d-1953252a73e9)

and Nature Remo-E lite;
![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/7629eeb2-8832-446b-8c55-87a0cac7fd9f)

**Comment by nao-pon** [comment]
@ahori Nature Remo E-Lite was reported in #121, but this integration cannot currently be supported. It seems that refrigerators can be supported by adding definitions to the [pychonet library](https://github.com/scottyphillips/pychonet). It will take some time, but we plan to fix it.

**Comment by nao-pon** [comment]
One way to obtain data from Japan's smart power meters via Route B is to use the "[For parts removal! Soc board with wireless module + white plastic case](https://akizukidenshi.com/catalog/g/g117437/)" currently sold by Akizuki Electronics. I'm testing it [here](https://github.com/nao-pon/python-echonet-lite/tree/dg-wisun) and it's working fine.

**Comment by ahori** [comment]
I have forked your pychonet lib and found fridge implementation in it.  I wonder why home assistant is unable to recognize my fridge.  Well, I am going to investigate and try to fix my issues, but I have no knowledge about how the HACS components are managed by home assistant.  I would be happy to investigate the problems if you let mw know how to test my fixes on pychonet on my home assistant. Sorry for this very stupid question.

**Comment by nao-pon** [comment]
I added the refrigerator class yesterday. It's still being fixed, so I'll let you know when it's available for testing. 😄

**Comment by nao-pon** [comment]
It's now testable on the edge branch of my repository. I can't test it, so could you please verify it?

Please refer to the following article to install the edge branch.
- https://github.com/scottyphillips/echonetlite_homeassistant/issues/161#issuecomment-1911288226

**Comment by ahori** [comment]
Wow! What a fast on-demond coding!! Thank you very much.

The new one looks much better, but something weird. 

![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/19c8cc9b-4925-4078-97b6-7c670b806d86)

Let's ignore the "Ma..." sensor (6th from the top) at this time, I know the echonet protocol is a kind of bull-shit.
This snapshot looks good but when I check each sensor status, for example, "Fridge Door Open Warning" at the top,

![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/f284076a-8958-4ba2-ae65-b56f93ae2d60)

its status is '42.' This must be something else, such as "False" or "No."

If you are kind and have some time, I would appreciate it if you let me know how the echonetlite_homeassistant repo includes the pychonet library.

**Comment by ahori** [comment]
I think https://github.com/scottyphillips/echonetlite_homeassistant/pull/183 would be the fix of above my comment.

**Comment by nao-pon** [comment]
I also merged #183 into my edge. What about the results?

**Comment by ahori** [comment]
Thanks again. It looks better, 

![image](https://github.com/scottyphillips/echonetlite_homeassistant/assets/16380686/f028f712-9243-45dc-9f1d-83e8ce25954b)

but I'm wondering if the "Maximum Allowed Value..." entity has any meaning and what the values of "101" are. According to the ECHONET specification (https://echonet.jp/wp/wp-content/uploads/pdf/General/Standard/Release/Release_R/Appendix_Release_R_r2.pdf), those values are _levels_, not temperature values. So, I think you may simply ignore them (not to confuse users). Or, just left them as they are now.  Please do not get me wrong, I do not want to ruin your great job.

**Comment by nao-pon** [comment]
Yes, I set those less useful entities to be disabled by default on installation.
However, if it is currently enabled, you must manually disable it.

I think the standard for Echonet-Lite itself is pretty good, but Japanese manufacturers are too closed-minded and have not adopted this standard (or even if they have adopted it, they have not actively publicized it). As a result, this standard has hardly become established and is no longer being adopted in new products. This is very unfortunate, and I think it has resulted in consumers' awareness of IoT falling far behind the rest of the world. That's such a shame...

**Comment by scottyphillips** [comment]
If I can comment from the point of view of someone living outside Japan, Echonet Lite is miles in front of comparable American or European standards. I think, however, a lot of tech companies across the globe have unfortunately adopted a view where shunting their customers into little non-standard ecosystem bubbles is good for their business because they get an opportunity to take ownership and monetise customer data in the cloud.

**Comment by nao-pon** [comment]
@scottyphillips I learned from your post that it's similar in America and Europe. I hope Matter improves this in the near future... Famous Japanese manufacturers are completely useless.

@ahori Version 3.8.2 now supports refrigerators, so I'll close this. thank you!

**Comment by ahori** [comment]
@scottyphillips and @nao-pan,
Scott's comment got me thinking about the cultural difference between Japan and western countries. It ends up with that  I think the difference lies in the DIY spirit. Many Japanese DIYers use some simple method to assemble something, while western DIYers are doing much more complicated and excellent jobs. I realized this when I went to a US DIY shop. For example, the recent right-to-repair movement does not originate in Japan because of this.

Now go back to the ECHONET standard. As @nao-pan said, Japanese manufacturers do not come to think for end-users to do something on their products. And, thus, they do not actively publish specifications. Further, most companies do not understand the spirit of free open-source software. They think that making their software (specification) open and free does not produce any money and they are afraid of something. It is the consequences of this that they are far behind in IT.

I am sorry for this long comment. It is my great pleasure for me to work with all of you. Thank you.

P.S.
I am now investigating why my REMO-E (lite) does not work as a power meter. From the view of your ECHNOETlite integration, it works as a controller, not a power meter. I will create another PR when I succeed.
